addons/mail/models/res_users.py

```diff
@@ -77,7 +77,7 @@ def _inverse_notification_type(self):
 
     @api.depends_context("uid")
     def _compute_can_edit_role(self):
-        self.can_edit_role = self.env["res.role"].has_access("write")
+        self.can_edit_role = self.env["res.role"].sudo(flag=False).has_access("write")
```

Comment:
It's strange to use a keyword argument for `sudo`, but otherwise the fix looks ok.

Reply:
Isn't it basically equivalent to calling `_check_access` instead, which doesn't check for the su flag?